examples/bind-shopping.amp.html

```diff
@@ -0,0 +1,454 @@
+            attr="out-of-stock"
+            toggle="{{!items[5].sizes.length}}">
+        </amp-bind-attribute>
+      </amp-img>
+
+      <amp-img src="./shirts/dark-green.jpeg" width=50 height=50 layout=fixed>
+        <amp-bind-event layout=nodisplay
+            on="tap"
+            variable="selected"
+            value="{{6}}">
+        </amp-bind-event>
+        <amp-bind-attribute layout=nodisplay
+            attr="selected"
+            toggle="{{selected === 6}}">
+        </amp-bind-attribute>
+        <amp-bind-attribute layout=nodisplay
+            attr="out-of-stock"
+            toggle="{{!items[6].sizes.length}}">
+        </amp-bind-attribute>
+      </amp-img>
+
+      <amp-img src="./shirts/light-gray.jpeg" width=50 height=50 layout=fixed>
+        <amp-bind-event layout=nodisplay
+            on="tap"
+            variable="selected"
+            value="{{7}}">
+        </amp-bind-event>
+        <amp-bind-attribute layout=nodisplay
+            attr="selected"
+            toggle="{{selected === 7}}">
+        </amp-bind-attribute>
+        <amp-bind-attribute layout=nodisplay
+            attr="out-of-stock"
+            toggle="{{!items[7].sizes.length}}">
+        </amp-bind-attribute>
+      </amp-img>
+
+      <amp-img src="./shirts/white.jpeg" width=50 height=50 layout=fixed>
+        <amp-bind-event layout=nodisplay
+            on="tap"
+            variable="selected"
+            value="{{8}}">
+        </amp-bind-event>
+        <amp-bind-attribute layout=nodisplay
+            attr="selected"
+            toggle="{{selected === 8}}">
+        </amp-bind-attribute>
+        <amp-bind-attribute layout=nodisplay
+            attr="out-of-stock"
+            toggle="{{!items[8].sizes.length}}">
+        </amp-bind-attribute>
+      </amp-img>
+    </div>
+
+    <amp-accordion>
+      <!--
+      <section expanded bind-attr="{expanded: !size}">
+      -->
+      <section expanded>
+        <amp-bind-attribute width=355 height=50
+            attr="expanded"
+            toggle="{{!size}}">
+
+          <!-- Nested here due to accordion's child limit. -->
+          <!--
+          <amp-img width=355 height=50 src="https://placehold.it/355x50?text=SIZE%20???"
+              bind-src="https://placehold.it/355x50?text=SIZE%20{{size}}">
+          -->
+          <amp-img width=355 height=50 src="https://placehold.it/355x50?text=SIZE%20???">
+            <amp-bind-attribute layout=nodisplay
+                attr="src"
+                value="{{'https://placehold.it/355x50?text=SIZE%20' + size}}">
+            </amp-bind-attribute>
+          </amp-img>
+
+        </amp-bind-attribute>
+
+        <div class="sizes">
+          <!--
+          <button
+              on="tap:setState({size: 'SMALL'})"
+              bind-class="{hidden: items[selected].sizes.indexOf('S') < 0}">
+          -->
+          <button>
+            <amp-bind-event layout=nodisplay
+                on="tap"
+                variable="size"
```

Comment:
Yes. The document has a single scope for local state, which is initialized with `<amp-bind-dataset>`. The assignment is done on that local state.